src/compiler/moduleNameResolver.ts

```diff
@@ -117,9 +117,10 @@ namespace ts {
         }
     }
 
-    function readJson(path: string, host: ModuleResolutionHost): PackageJson {
+    /* @internal */
+    export function readJson(path: string, host: { readFile(fileName: string): string | undefined }): object {

```

Comment:
Why not mark `readFile` optional?

Reply:
Whoops, left in a change to the function body on accident. This should not be optional here because it is checked by the callers.